src/boot/meson.build

```diff
@@ -74,6 +74,12 @@ if conf.get('ENABLE_BOOTLOADER') != 1
 endif
 
 efi_conf = configuration_data()
+
+# import several configs from userspace
+foreach name : ['HAVE_WARNING_ZERO_LENGTH_BOUNDS', 'HAVE_WARNING_ZERO_AS_NULL_POINTER_CONSTANT']
+        efi_conf.set10(name, conf.get(name) == 1)
```

Comment:
== 1 seems unnecessary?

Reply:
Meson wants a boolean, not an integer. So yeah, we need the explicit comparison.

Reply:
In general, comparing `== true` never makes sense in meson, since `if true == true` is folded and evaluated as `if true`.

Comparing `== 1` is logical because the only time it is allowed to use this is when the left-hand-side is an integer, and the return value of `==` is a boolean, and meson doesn't generally accept numbers and truthiness-values interchangeably.